src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -223,20 +223,18 @@ class Peer(
         launch {
             swapInWallet.walletStateFlow
                 .filter { it.consistent }
-                .fold(false) { swapAlreadyAttempted, wallet ->
-                    val balance = wallet.confirmedBalance
+                .fold(emptySet<WalletState.Utxo>()) { reservedUtxos, wallet ->
+                    // reservedUtxos are part of a previously issued RequestChannelOpen command
+                    val availableWallet = wallet.minus(reservedUtxos)
+                    val balance = availableWallet.confirmedBalance
                     if (balance > 10_000.sat) {
-                        if (swapAlreadyAttempted) {
-                            logger.info { "$balance available on swap-in wallet but swap-in already attempted: not doing anything" }
-                        } else {
-                            logger.info { "$balance available on swap-in wallet: requesting channel" }
-                            input.send(RequestChannelOpen(Lightning.randomBytes32(), wallet, maxFeeBasisPoints = 100, maxFeeFloor = 3_000.sat)) // 100 bips = 1 %
-                        }
-                        true
+                        logger.info { "$balance available on swap-in wallet: requesting channel" }
+                        input.send(RequestChannelOpen(Lightning.randomBytes32(), availableWallet, maxFeeBasisPoints = 100, maxFeeFloor = 3_000.sat)) // 100 bips = 1 %
+                        reservedUtxos.union(availableWallet.confirmedUtxos)
                     } else {
-                        logger.info { "$balance available on swap-in wallet but amount insufficient: waiting for more" }
-                        swapAlreadyAttempted
-                    }
+                        logger.info { "insufficient balance on swap-in wallet ($balance, ${availableWallet.unconfirmedBalance} unconfirmed): waiting for more" }
```

Comment:
It's gonna sound like I'm bikeshedding, but the issue I have with this log is that it makes it implies there is something wrong with having 0.sat ("insufficient balance"), whereas this is not only normal, but the most common case.

Reply:
Ah, I didn't understand what you meant initially. That makes sense.

I took another stab at it in [a9be848](https://github.com/ACINQ/lightning-kmp/pull/397/commits/a9be84817ffa36c4716ec4ac18eb93b782e87b40). I actually didn't like the fact that it only logged the full balance (confirmed + unconfirmed) in the `else` branch. So I moved the logs around:

```kotlin
val balance = availableWallet.confirmedBalance
logger.info { "swap-in wallet balance: $balance, ${availableWallet.unconfirmedBalance} unconfirmed" }
if (balance > 10_000.sat) {
    logger.info { "swap-in wallet: requesting channel using confirmed balance: $balance" }
    // ... code ...
} else {
    // ... code ...
}
```

What do you think ?

Reply:
👍